google/cloud/speech/v1/internal/adaptation_logging_decorator.h

```diff
@@ -0,0 +1,99 @@
+
+// Generated by the Codegen C++ plugin.
+// If you make any local changes, they will be lost.
+// source: google/cloud/speech/v1/cloud_speech_adaptation.proto
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_SPEECH_V1_INTERNAL_ADAPTATION_LOGGING_DECORATOR_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_SPEECH_V1_INTERNAL_ADAPTATION_LOGGING_DECORATOR_H
+
+#include "google/cloud/speech/v1/internal/adaptation_stub.h"
+#include "google/cloud/tracing_options.h"
+#include "google/cloud/version.h"
+#include <memory>
+#include <set>
+#include <string>
+
+namespace google {
+namespace cloud {
+namespace speech_v1_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+class AdaptationLogging : public AdaptationStub {
+ public:
+  ~AdaptationLogging() override = default;
+  AdaptationLogging(std::shared_ptr<AdaptationStub> child,
+                    TracingOptions tracing_options,
+                    std::set<std::string> components);
+
+  StatusOr<google::cloud::speech::v1::PhraseSet> CreatePhraseSet(
+      grpc::ClientContext& context,
+      google::cloud::speech::v1::CreatePhraseSetRequest const& request)
+      override;
+
+  StatusOr<google::cloud::speech::v1::PhraseSet> GetPhraseSet(
+      grpc::ClientContext& context,
+      google::cloud::speech::v1::GetPhraseSetRequest const& request) override;
+
+  StatusOr<google::cloud::speech::v1::ListPhraseSetResponse> ListPhraseSet(
+      grpc::ClientContext& context,
+      google::cloud::speech::v1::ListPhraseSetRequest const& request) override;
+
+  StatusOr<google::cloud::speech::v1::PhraseSet> UpdatePhraseSet(
+      grpc::ClientContext& context,
+      google::cloud::speech::v1::UpdatePhraseSetRequest const& request)
+      override;
+
+  Status DeletePhraseSet(
+      grpc::ClientContext& context,
+      google::cloud::speech::v1::DeletePhraseSetRequest const& request)
+      override;
+
+  StatusOr<google::cloud::speech::v1::CustomClass> CreateCustomClass(
+      grpc::ClientContext& context,
+      google::cloud::speech::v1::CreateCustomClassRequest const& request)
+      override;
+
+  StatusOr<google::cloud::speech::v1::CustomClass> GetCustomClass(
+      grpc::ClientContext& context,
+      google::cloud::speech::v1::GetCustomClassRequest const& request) override;
+
+  StatusOr<google::cloud::speech::v1::ListCustomClassesResponse>
+  ListCustomClasses(grpc::ClientContext& context,
+                    google::cloud::speech::v1::ListCustomClassesRequest const&
+                        request) override;
+
+  StatusOr<google::cloud::speech::v1::CustomClass> UpdateCustomClass(
+      grpc::ClientContext& context,
+      google::cloud::speech::v1::UpdateCustomClassRequest const& request)
+      override;
+
+  Status DeleteCustomClass(
+      grpc::ClientContext& context,
+      google::cloud::speech::v1::DeleteCustomClassRequest const& request)
+      override;
+
+ private:
+  std::shared_ptr<AdaptationStub> child_;
+  TracingOptions tracing_options_;
+  std::set<std::string> components_;
```

Comment:
You'll need to rebase/regenerate now that I managed to get #12172 in first.

Reply:
Yep. And I think we are going to see failures from `retail` soon :upside_down_face:

Reply:
I'll fix it up.

Reply:
Rebased and regenerated.